lib/collection.js

```diff
@@ -280,7 +280,7 @@ const DEPRECATED_FIND_OPTIONS = ['maxScan', 'fields', 'snapshot'];
  */
 const findSchema = {
   limit: { type: 'number', default: 0 },
-  sort: { type: ['array', 'object'] },
+  sort: { type: [Array, 'object'] },
```

Comment:
Arrays are a special case in type comparisons. Instead of relying on `instanceof Array`, we should use `Array.isArray`. I think it would probably be a good idea to treat `'array'` as a special type at https://github.com/mongodb/node-mongodb-native/pull/1859/files#diff-e1fa59f7e4e4c9080c24be96453dfdb8R154 where we compare types. Something like:

```js
if (optionType === 'array') {
  if (!Array.isArray(optionValue)) {
    invalidate(validationLevel, invalidateMessageType, logger);
  }
}
```

Reply:
👍 with this idea